pkg/cmd/pipeline/start.go

```diff
@@ -0,0 +1,174 @@
+		if len(args) == 0 {
+			return errNoPipeline
+		}
+
+		c, err := p.Clients()
+		if err != nil {
+			return err
+		}
+
+		name, ns := args[0], p.Namespace()
+		_, err = c.Tekton.TektonV1alpha1().Pipelines(ns).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return errInvalidPipeline
+		}
+
+		return nil
+	}
+
+}
+
+func startCommand(p cli.Params) *cobra.Command {
+	var (
+		res    []string
+		params []string
+		svc    string
+	)
+
+	c := &cobra.Command{
+		Use:     "start",
+		Aliases: []string{"trigger"},
+		Short:   "Start pipelines",
+		Example: `
+  # start pipeline foo by creating a pipelienrun named "foo-run-xyz123" from the namespace "bar"
+    tkn pipeline start foo --param NAME=VALUE --resource source=scaffold-git  -s ServiceAccountName  -n bar
+   `,
+		SilenceUsage: true,
+		Args:         NameArg(p),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			opt := triggerOptions{
+				Resources:          res,
+				Params:             params,
+				ServiceAccountName: svc,
+			}
+			return trigger(cmd.OutOrStdout(), opt, p, args[0])
+		},
+	}
+
+	c.Flags().StringSliceVarP(&res, "resource", "r", []string{}, "pass the resource name and ref")
+	c.Flags().StringSliceVarP(&params, "param", "p", []string{}, "pass the param")
+	c.Flags().StringVarP(&svc, "serviceaccount", "s", svc, "pass the serviceaccount name")
+
+	return c
+}
+
+func trigger(out io.Writer, opt triggerOptions, p cli.Params, pName string) error {
+
+	cs, err := p.Clients()
+	if err != nil {
+		return err
+	}
+
+	res, err := parseRes(opt.Resources)
+	if err != nil {
+		return err
+	}
+
+	params, err := parseParam(opt.Params)
+	if err != nil {
+		return err
+	}
+
+	pr := &v1alpha1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    p.Namespace(),
+			GenerateName: pName + "-run-",
+		},
+		Spec: v1alpha1.PipelineRunSpec{
+			PipelineRef:    v1alpha1.PipelineRef{Name: pName},
+			ServiceAccount: opt.ServiceAccountName,
+			Resources:      res,
+			Params:         params,
+		},
+	}
+
+	pr, err = cs.Tekton.TektonV1alpha1().PipelineRuns(p.Namespace()).Create(pr)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(out, "Pipelinerun started: %s\n", pr.Name)
+	return nil
+}
+
+func parseRes(res []string) ([]v1alpha1.PipelineResourceBinding, error) {
+	var resources []v1alpha1.PipelineResourceBinding
+	for _, v := range res {
+		r := strings.Split(v, "=")
+		if len(r) != 2 {
+			errMsg := errInvalidResource.Error() + v +
+				"\n Please pass resource as -p ResourceName=ResourceRef"
+			return nil, errors.New(errMsg)
+		}
+		resources = append(resources, v1alpha1.PipelineResourceBinding{
+			Name: r[0],
+			ResourceRef: v1alpha1.PipelineResourceRef{
+				Name: r[1],
+			},
+		})
+	}
+	return resources, nil
+}
+
+func parseParam(p []string) ([]v1alpha1.Param, error) {
+	var params []v1alpha1.Param
+	for _, v := range p {
+		r := strings.Split(v, "=")
+		if len(r) != 2 {
+			errMsg := errInvalidParam.Error() + v +
```

Comment:
why define var errInvalidParam as error if the idea is to convert it into a string?

Reply:
Changed it to string. Earlier I was just returning that error and later decided to add parameter